### PR TITLE
Add script to release to git tags

### PR DESCRIPTION
**Pull request by hedgepigdaniel**
## Summary

When making a private fork to develop a change, it is difficult to test the change with another project. `yarn link` works but requires extra steps each time the project is checked out. Installing from git doesn't work easily because the build assets are not in the repository and because the repository is a monorepo.

This adds a script modelled after https://github.com/respond-framework/rudy/pull/34 which makes a git tag containing the built version of each package that has changed since the last release (as it would be published to NPM).

My fork of `npm-publish-git` is used because of https://github.com/andreineculau/npm-publish-git/pull/4

## Test plan

I created an example release containing the changes from https://github.com/gregberge/loadable-components/pull/483:
- https://github.com/hedgepigdaniel/loadable-components/releases/tag/%40loadable%2Fcomponent%2Fbuild%2Fnamed-exports%2F6.0.0-from-git.0
- https://github.com/hedgepigdaniel/loadable-components/releases/tag/%40loadable%2Fbabel-plugin%2Fbuild%2Fnamed-exports%2F6.0.0-from-git.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +48,7 @@
     "eslint-plugin-react": "^7.17.0",
     "jest": "^24.9.0",
     "lerna": "^3.19.0",
+    "npm-publish-git": "git://github.com/hedgepigdaniel/npm-publish-git.git",
```
Comment (theKashey) [inline package.json]: what about `@hedgepigdaniel/npm-publish-git`?

Comment (hedgepigdaniel) [reply package.json]: :+1: it has been done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -61,6 +61,12 @@ _Before_ submitting a pull request, please make sure the following is done…
     $ yarn test
     ```
 
+#### Testing with your own project
+
+You can use `yarn run release-from-git` to create releases as tags on github. This requires that:
+ - Your git remote (where you want to publish the tags) is `origin`
+ - Your commit messages follow the conventional commits specification
```
Comment (theKashey) [inline CONTRIBUTING.md]: this is not quite clear. A minimal example would help a lot.

Comment (hedgepigdaniel) [reply CONTRIBUTING.md]: I added a minimal example and a link to the spec

**Pull request merged**